src/pydicom/encaps.py

```diff
@@ -803,3 +912,421 @@ def encapsulate_extended(frames: list[bytes]) -> tuple[bytes, bytes, bytes]:
+        All fragments as a list of ``bytes``.
+    """
+    # Convert data into a memory-mapped file
+    with DicomBytesIO(data) as fp:
+        # DICOM standard requires this
+        fp.is_little_endian = True
+        BasicOffsetTable = _read_item(fp)  # NOQA
+        seq = []
+
+        while True:
+            item = _read_item(fp)
+
+            # None is returned if get to Sequence Delimiter
+            if not item:
+                break
+            seq.append(item)
+
+        # XXX should
+        return seq
+
+
+def _defragment_data(data: bytes) -> bytes:
+    """Read encapsulated data and return the fragments as one continuous bytes.
+
+    .. deprecated:: 3.0
+
+        This function will be removed in v4.0, Please use
+        :func:`~pydicom.encaps.generate_frames` for generating frame
+        data or :func:`~pydicom.encaps.generate_fragments` for generating
+        fragment data.
+
+    Parameters
+    ----------
+    data : bytes
+        The encapsulated pixel data fragments.
+
+    Returns
+    -------
+    bytes
+        All fragments concatenated together.
+    """
+    return b"".join(_decode_data_sequence(data))
+
+
+def _read_item(fp: DicomIO) -> bytes | None:
+    """Read and return a single Item in the fragmented data stream.
+
+    .. deprecated:: 3.0
+
+        This function will be removed in v4.0, please use
+        :func:`~pydicom.encaps.generate_fragments` instead.
+
+    Parameters
+    ----------
+    fp : filebase.DicomIO
+        The file-like to read the item from.
+
+    Returns
+    -------
+    bytes
+        The Item's raw bytes.
+    """
+
+    logger = config.logger
+    try:
+        tag = fp.read_tag()
+
+    # already read delimiter before passing data here
+    # so should just run out
+    except EOFError:
+        return None
+
+    # No more items, time for sequence to stop reading
+    if tag == SequenceDelimiterTag:
+        length = fp.read_UL()
+        logger.debug("%04x: Sequence Delimiter, length 0x%x", fp.tell() - 8, length)
+
+        if length != 0:
+            logger.warning(
+                "Expected 0x00000000 after delimiter, found 0x%x,"
+                " at data position 0x%x",
+                length,
+                fp.tell() - 4,
+            )
+        return None
+
+    if tag != ItemTag:
+        logger.warning(
+            "Expected Item with tag %s at data position 0x%x", ItemTag, fp.tell() - 4
+        )
+        length = fp.read_UL()
+    else:
+        length = fp.read_UL()
+        logger.debug("%04x: Item, length 0x%x", fp.tell() - 8, length)
+
+    if length == 0xFFFFFFFF:
+        raise ValueError(
+            "Encapsulated data fragment had Undefined Length"
+            f" at data position 0x{fp.tell() - 4:x}"
+        )
+
+    item_data = fp.read(length)
+    return item_data
+
+
+_DEPRECATED = {
+    "get_frame_offsets": _get_frame_offsets,
+    "get_nr_fragments": _get_nr_fragments,
+    "generate_pixel_data_fragment": _generate_pixel_data_fragment,
```

Comment:
what about `itemise_frame` => `itemize_frame`?

Reply:
I just removed it, it's an alias and it's not in the API reference so I figured it'd be OK. I've added it to the `_DEPRECATED` dict anyway.